App/Dependencies/ExposureDetection/ImmuniTemporaryExposureKeyProvider.swift

```diff
@@ -49,18 +57,36 @@ class ImmuniTemporaryExposureKeyProvider: TemporaryExposureKeyProvider {
       .delete(chunks.flatMap { $0.localUrls })
   }
 
-  private func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
+  func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
     self.networkManager.getKeysIndex()
       .then { (keysIndex: KeysIndex) -> [Int] in
-        guard let latestKnownChunkIndex = latestKnownChunkIndex else {
-          return Array(keysIndex.oldest ... keysIndex.newest)
-        }
 
-        guard keysIndex.newest > latestKnownChunkIndex else {
+        let latestKnown = latestKnownChunkIndex ?? -1
```

Comment:
I liked the older version with the guard better, but maybe I'm biased. :D

Reply:
I added it because otherwise you need to check the cap twice